src/declarations/stencil-public-compiler.ts

```diff
@@ -2265,17 +2265,11 @@ export interface Diagnostic {
   messageText: string;
   debugText?: string;
   code?: string;
-  absFilePath?: string;
-  relFilePath?: string;
-  lineNumber?: number;
-  columnNumber?: number;
-  lines?: {
-    lineIndex: number;
-    lineNumber: number;
-    text?: string;
-    errorCharStart: number;
-    errorLength?: number;
-  }[];
+  absFilePath?: string | undefined;
+  relFilePath?: string | undefined;
+  lineNumber?: number | undefined;
+  columnNumber?: number | undefined;
+  lines: PrintLine[];
```

Comment:
does this count as the type of change where we'd need to make it a breaking change? if so I'll close this PR and open against the v3 dev branch instead

Reply:
Yeah, I think we'd need to target v3.0.0 since this is in `stencil-public-compiler` (more specifically, is exported by it). I don't have a list on hand of who is using `Diagnostic` in their APIs, but I wouldn't be surprised if some nx-related tooling was. I'd target v3 to be on the safe side

Reply:
that makes sense, just updated it to be against `v3.0.0-dev` instead